_posts/2018-05-23-gracefully_clean_up_in_grpc_junit_tests.md

```diff
@@ -0,0 +1,111 @@
+---
+layout: post
+title: Let's gracefully clean up in gRPC JUnit tests
+published: false
+permalink: blog/gracefully_clean_up_in_grpc_junit_tests
+author: Dapeng Zhang
+company: Google
+company-link: https://www.google.com
+---
+
+As a good programmer, it's important that you remember to always clean up gRPC resources such as client channels, servers, and previously attached Contexts whenever they are no longer needed. This is even true for JUnit tests, because otherwise leaked resource will not only linger in your machine forever in some cases but also interfere with subsequent tests. A not-so-bad case is that subsequent tests can not pass because of the leaked resource in the previous test. The worst case is that some subsequent tests pass but in fact they wouldn't have passed if the previously passed test had not leaked resource.
+
+<!--more-->
+
+So cleanup, cleanup, cleanup...\
```

Comment:
I am not sure what the trailing backslash does here.   Is this markdown?

Reply:
This is a line break in markdown
https://spec.commonmark.org/0.28/#hard-line-break